src/mcradar/zeOperator.py

```diff
@@ -182,47 +182,94 @@ def calcParticleZe(wls, elvs,mcTable, mcTableAgg,mcTableCry,scatSet,beta,beta_st
+                        elevation 	= np.ones(len(mcTableAgg.mTot.values))*elv,
+                        wavelength  = np.ones(len(mcTableAgg.mTot.values))*wl,
+                    )
                 search_idx = search_ckdtree(tree, scaling, target)
-                                
-                for i_part, trgt, idx in (zip(range(len(mcTableAgg)), zip(*target.values()), search_idx)):
-                    #print(f"for super particle {trgt=}, we have {len(idx)=} entries") 
-                    scatPoints = DDA_data_agg.isel(index=idx)
-
-                    superparticle = mcTableAgg.isel(index=i_part)
-                    if len(scatPoints.index) > superparticle.sMult.values:
-                        idx_sel = np.random.choice(np.arange(0,len(scatPoints.index)-1), size = int(superparticle.sMult.values),replace=False)
-                    elif len(scatPoints.index)==0:
-                        print('Warning: no scatPoints found for superparticle {0}, wl {1}, elv {2}'.format(superparticle.index.values, wl,elv))
-                        print('dia',superparticle.dia.values, 'mass',superparticle.mTot.values)
-                        continue
-                    else:
-                        print('Warning: not enough scatPoints, sMult {0}, scatPoints {1}'.format(superparticle.sMult.values, len(scatPoints.index)))
-                        print(superparticle.dia.values, superparticle.mTot.values, superparticle.index.values, wl,elv)
-                        idx_sel = np.random.choice(np.arange(0,len(scatPoints.index)-1), size = int(superparticle.sMult.values),replace=True)
-                        
-                    # we can already sum the randomly selected points here, because otherwise we would have summed them for creating the Doppler spectra anyway. We just need to make sure that now the multiplicity is one now, so that this particle is only taken once into account for spectrum.
-                    mcTable['sZeH'].loc[elv,wl,superparticle.index] = scatPoints['ZeH'][idx_sel].values.sum()*2*np.pi 
-                    mcTable['sCextH'].loc[elv,wl,superparticle.index] = scatPoints['CextH'][idx_sel].values.sum()*2*np.pi 
-                    mcTable['sCextV'].loc[elv,wl,superparticle.index] = scatPoints['CextV'][idx_sel].values.sum()*2*np.pi 
-                    mcTable['sZeV'].loc[elv,wl,superparticle.index] = scatPoints['ZeV'][idx_sel].values.sum()*2*np.pi 
-                    mcTable['sZeHV'].loc[elv,wl,superparticle.index] = scatPoints['ZeHV'][idx_sel].values.sum()*2*np.pi 
-                    mcTable['sKDP'].loc[elv,wl,superparticle.index] = scatPoints['KDP'][idx_sel].values.sum()*2*np.pi 
-                    mcTable['sMult'].loc[superparticle.index] = 1
-                    
-                    
-                    
+
+                if False:
+                    for i_part, trgt, idx in tqdm(zip(range(mcTableAgg.index.size), zip(*target.values()), search_idx), total=mcTableAgg.index.size):
```

Comment:
note here the change with respect to i_part. before it was len(mcTableAgg). This counted only 33 data variables but I think should have been superparticles